.github/workflows/release_fips.yaml

```diff
@@ -0,0 +1,182 @@
+              fi
+            fi
+          done
+
+          # Convert to JSON array for matrix output
+          MATRIX_JSON=$(echo "$FILTERED_MATRIX" | jq -R -s -c 'split("\n") | map(select(length > 0)) | {version: .}')
+
+          echo "matrix=$MATRIX_JSON" >> $GITHUB_OUTPUT
+
+  build-k8s-versions:
+    needs: generate-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJson(needs.generate-matrix.outputs.matrix) }}
+      fail-fast: false
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build kubernetes FIPS ${{ matrix.version }}
+        run: |
+          # Setup GitHub CLI
+          gh auth login --with-token <<< "${{ secrets.GITHUB_TOKEN }}"
+          
+          FORCE="${{ inputs.force }}"
+
+          cd fips
+
+          VERSION=${{ matrix.version }}
+          KUBERNETES_VERSION_TRIMMED=$(echo ${VERSION} | sed -E 's/^(v[0-9]+\.[0-9]+)\.[0-9]+$/\1/')
+          
+          echo "Building FIPS release for Kubernetes ${VERSION}.."
+      
+          # Download for amd64 and arm64
+          ./build-fips.sh --target-arch amd64 --kubernetes-version ${VERSION} --versions-file "../hack/kubernetes-${KUBERNETES_VERSION_TRIMMED}"       
+          ./build-fips.sh --target-arch arm64 --kubernetes-version ${VERSION} --versions-file "../hack/kubernetes-${KUBERNETES_VERSION_TRIMMED}"
+      
+          # Create a new release with the archives as assets
+          if gh release view $VERSION --repo ${{ github.repository }} &>/dev/null; then
+            gh release upload "$VERSION" \
+              "./kubernetes-$VERSION-amd64-fips.tar.gz" \
+              "./kubernetes-$VERSION-arm64-fips.tar.gz" \
+              "./kubernetes-$VERSION-amd64-fips-full.tar.gz" \
+              "./kubernetes-$VERSION-arm64-fips-full.tar.gz" \
+              --repo ${{ github.repository }} \
+              --clobber
+
+            # Edit the release notes to include the component versions
+            gh release edit "$VERSION" \
+              --notes-file "./kubernetes-$VERSION.txt" \
+              --repo ${{ github.repository }}
+
+            echo "Updated release for Kubernetes $VERSION"
+          else
+            gh release create $VERSION \
+              --title "Kubernetes $VERSION" \
+              --notes-file "./kubernetes-$VERSION.txt" \
+              --repo ${{ github.repository }} \
+              "./kubernetes-$VERSION-amd64-fips.tar.gz" \
+              "./kubernetes-$VERSION-arm64-fips.tar.gz" \
+              "./kubernetes-$VERSION-amd64-fips-full.tar.gz" \
+              "./kubernetes-$VERSION-arm64-fips-full.tar.gz"
+      
+            echo "Released Kubernetes $VERSION"
```

Comment:
Lets fail here since we only want to add fips images to existing releases but not create them in this pipeline